File: src/dataset/javascript/isogram_exercism-javascript-bearguns/openAI/ChatGPT4_isogram_v1.js
var Isogram = function(string) {
  this.string = string;
  this.isIsogram = function() {
    const cleanedString = this.string.toLowerCase().replace(/[\s-,\.]/g, '');
    const seen = new Set();

    for (const char of cleanedString) {
      if (seen.has(char)) {
        return false;
      }
      seen.add(char);
    }

    return true;
  };
};

export default Isogram;

File: src/dataset/go/isogram_Exercism_go_rootulp/openAI/ChatGPT4_isogram_v1.go
package isogram

import (
	"unicode"
)

// IsIsogram returns whether the provided string is an isogram.
// In other words, whether the string does not contain any duplicate characters.
func IsIsogram(s string) bool {
	seen := make(map[rune]bool)

	for _, r := range s {
		if unicode.IsLetter(r) {
			r = unicode.ToLower(r)
			if seen[r] {
				return false
			}
			seen[r] = true
		}
	}
	return true
}